core/Azure.Mcp.Core/src/Areas/Server/Commands/ServiceStartCommand.cs

```diff
@@ -382,8 +544,15 @@ private static void ConfigureMcpServer(IServiceCollection services, ServiceStart
     /// Gets a safe ASP.NET Core URL with security validation.
     /// </summary>
     /// <returns>A validated URL string for ASP.NET Core binding.</returns>
-    private static string GetSafeAspNetCoreUrl()
+    private static string? GetSafeAspNetCoreUrl(ServiceStartOptions options)
     {
+        if (options.RunAsRemoteHttpService)
+        {
+            // When running as a remote HTTP service, let the typical IConfiguration
+            // binding handle the ASPNETCORE_URLS value without additional validation.
+            return null;
+        }
```

Comment:
If we make this change ([link](https://github.com/microsoft/mcp/pull/910/files#r2450058269)), we can keep this method as is and return string instead of string?.

Reply:
I think I'm going to update `GetSafeAspNetCoreUrl` to take in the builder so that it can encapsulate both possible paths then return `void`.

Reply:
Pushed. Please take a look. Marking as resolved.